### PR TITLE
chore: add talisman env var to config

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
This allows us to pass the talisman config from the ephemeral docker image environment variable into the config so that we can turn off talisman for ephemerals.

The Talisman os env var for the ephemeral instance was added here: https://github.com/apache/superset/pull/24627 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base.py [hunk 1]
```diff
@@ -120,6 +120,7 @@
     "ALERT_REPORTS_DEFAULT_RETENTION",
     "ALERT_REPORTS_DEFAULT_WORKING_TIMEOUT",
     "NATIVE_FILTER_DEFAULT_ROW_LIMIT",
+    "TALISMAN_ENABLED",
```
[the remark this answers is not in the document]
Comment (eschutho) [reply superset/views/base.py:123]: Sorry for the confusion. I just marked the PR as draft again. I'm just using that as a test to see if the config is being applied in the ephemeral environment or not.

**Comment by eschutho** [comment]
@michael-s-molina it looks like it got this to work, if you want to review. I just need to remove my test commit.

File: superset/views/base.py [hunk 2]
```diff
@@ -450,6 +450,7 @@ def cached_common_bootstrap_data(user: User, locale: str) -> dict[str, Any]:
         "extra_categorical_color_schemes": conf["EXTRA_CATEGORICAL_COLOR_SCHEMES"],
         "theme_overrides": conf["THEME_OVERRIDES"],
         "menu_data": menu_data(user),
+        "Talisman": conf.get("TALISMAN_ENABLED", "not set"),
```
Comment (eschutho) [inline superset/views/base.py:453]: I need to remove this.

Comment (eschutho) [inline superset/views/base.py:453]:
```suggestion
```

**Review by michael-s-molina — APPROVED** [review]
Nice! Thanks for working on this and for the added tests!

**Pull request merged**